src/libs/actions/IOU.ts

```diff
@@ -1740,9 +1740,18 @@ function getDeleteTrackExpenseInformation(
     if (shouldDeleteTransactionThread) {
         optimisticData.push(
             {
-                onyxMethod: Onyx.METHOD.SET,
+                onyxMethod: Onyx.METHOD.MERGE,
                 key: `${ONYXKEYS.COLLECTION.REPORT}${transactionThreadID}`,
-                value: null,
+                value: {
+                    reportID: null,
+                    stateNum: CONST.REPORT.STATE_NUM.APPROVED,
+                    statusNum: CONST.REPORT.STATUS_NUM.CLOSED,
+                    participants: {
+                        [userAccountID]: {
+                            notificationPreference: CONST.REPORT.NOTIFICATION_PREFERENCE.HIDDEN,
+                        },
+                    },
+                },
```

Comment:
Why are we changing this? When the transaction is deleted we should delete the transaction thread, no?

Reply:
@luacmartins We switched from using `SET` to `MERGE` to prevent the appearance of the not found page due to platforms updating the value too quickly. This change has already been made for [`leaveRoom`](https://github.com/Expensify/App/blob/7cb72604f0129e2771ac72bd83d1b33e9b80d594/src/libs/actions/Report.ts#L2971-L2980) and [`leaveGroupChat`](https://github.com/Expensify/App/blob/7cb72604f0129e2771ac72bd83d1b33e9b80d594/src/libs/actions/Report.ts#L2898-L2905) to address a similar issue https://github.com/Expensify/App/issues/49468.

Reply:
Thanks for clarifying. I think that makes sense, it just feels a bit odd to do that. Let's add a comment explaining why we're doing it so people don't try to revert this change

Reply:
done